helix-view/src/editor.rs

```diff
@@ -1033,6 +1072,8 @@ pub struct Editor {
     pub debugger_events: SelectAll<UnboundedReceiverStream<dap::Payload>>,
     pub breakpoints: HashMap<PathBuf, Vec<Breakpoint>>,
 
+    pub old_file_locs: HashMap<PathBuf, (ViewPosition, Selection)>,
```

Comment:
If I understand correctly, this `HashMap` will store the information of all the files opened by any instance of Helix in the system. Isn't that too much? Shouldn't a particular instance care only about its current working directory?

Reply:
Too much in terms of what? Memory footprint? With the default settings running for a few months I just checked my `file_history` file and it's 12KiB, when deserialised to a hash map it should be about the same size in RAM. That seems pretty negligible.

Personally I find this useful, as I do sometimes open files outside the working directory, and I don't see that there's really a tradeoff for it.

Reply:
Thanks for your answer. Yes, I meant memory footprint.

And it makes sense for it to be that low. After reading your answer I did some back of the envelop calculation and assuming a 64 bytes entry per file, you'll need 16K files in the history to crate a 1MiB footprint. Since we are pruning that history from time to time (IIUC), then it's unlikely we get to that extreme anyway.